File: zero_matrix.cpp
#include <iostream>

using namespace std;

void zero_matrix(int **matrix,int x,int y){

	int rowArray[x]={0,0,0,0}, colArray[y]={0,0,0,0,0};

	for(int i=0; i<x; i++){
		for(int j=0; j<y; j++){
			if(matrix[i][j] == 0){
				rowArray[i] = 1;
				colArray[j] = 1;
			}
		}
	}

	for(int i=0; i<x; i++){
		if(rowArray[i] == 1){
			for(int z=0; z<y; z++){
				matrix[i][z] = 0;
			}
		}
	}
	
	for(int j=0; j<y; j++){
		if(colArray[j] == 1){
			for(int z=0; z<x; z++){
				matrix[z][j] = 0;
			}
		}
	}
}

int main(){

	int x=4, y=5;
	int **matrix;

	matrix = new int *[x];
	for(int i=0; i<x; i++){
		matrix[i] = new int [y]; 
		for(int j=0; j<y; j++){
			matrix[i][j] = i+j;
		}
	}

	for(int i=0; i<x; i++){
		for(int j=0; j<y; j++){
			cout<< matrix[i][j] << "\t";
		}
		cout<< endl;
	}

	cout<< endl;
	
	zero_matrix(matrix,x,y);

	for(int i=0; i<x; i++){
		for(int j=0; j<y; j++){
			cout<< matrix[i][j] << "\t";
		}
		cout<< endl;
	}
	return 0;
}

File: zero_matrix.py

def zero_matrix(matrix,len1,len2):

	rowList = []
	for i in xrange(len1):
		rowList.append(0)

	colList = []
	for j in xrange(len2):
		colList.append(0)

	for i in xrange(len1):
		for j in xrange(len2):
			if matrix[i][j] == 0:
				rowList[i] = 1
				colList[j] = 1

	for i in xrange(len1):
		if (rowList[i] == 1):
			for z in xrange(len2):
				matrix[i][z] = 0

	for j in xrange(len2):
		if (colList[j] == 1):
			for z in xrange(len1):
				matrix[z][j] = 0

	return True

def main():

	len1 = 4
	len2 = 5

	matrix = [[0 for _ in xrange(len2)]for _ in xrange(len1)]

	for i in xrange(len1):
		for j in xrange(len2):
			matrix[i][j] = i + j

	matrix[3][2] = 0

	for i in xrange(len1):
		t = ""
		for j in xrange(len2):
			t = t + str(matrix[i][j]) + "\t"
		print(t)

	print("")

	zero_matrix(matrix,len1,len2)

	for i in xrange(len1):
		t = ""
		for j in xrange(len2):
			t = t + str(matrix[i][j]) + "\t"
		print(t)

if __name__ == "__main__":
	main()
